07-HTML5 APIs/js/canvas.js

```diff
@@ -0,0 +1,166 @@
+
+
+var app = angular.module('canvas', []);
+
+app.controller('Ctrl', function($scope) {
+
+//rectangle
+var canvas = document.getElementById('myCanvas');
+var context = canvas.getContext('2d');
+
+context.beginPath();
+context.rect(20, 50, 200, 100);
+context.fillStyle = '#'+Math.floor(Math.random()*16777215).toString(16);
+context.fill();
+context.lineWidth = 10;
+context.strokeStyle = '#'+Math.floor(Math.random()*16777215).toString(16);
+context.stroke();
+
+
+//circle
+
+var centerX = canvas.width / 2;
+var centerY = canvas.height / 2;
+var radius = 70;
+
+context.beginPath();
+context.arc(centerX, centerY, radius, 0, 2 * Math.PI, false);
+context.fillStyle = '#'+Math.floor(Math.random()*16777215).toString(16);
+context.fill();
+context.lineWidth = 10;
+context.strokeStyle = '#'+Math.floor(Math.random()*16777215).toString(16);
+context.stroke();
+
+//triangle
+
+context.fillStyle='#'+Math.floor(Math.random()*16777215).toString(16);
+context.beginPath();
+context.moveTo(100,410);
+context.lineTo(200,200);
+context.lineTo(300,410);
+context.closePath();
+context.fill();
+context.lineWidth = 10;
+context.strokeStyle = '#'+Math.floor(Math.random()*16777215).toString(16);
+context.stroke();
+
+//mio
+(function(){
+	var c = document.getElementById('myCanvas'),
+	ctx = c.getContext('2d');
+
+	function draw (){
+
+		context.fillStyle="blue";
+		context.beginPath();
+		context.moveTo(100,410);
+		context.lineTo(200,200);
+		context.lineTo(300,410);
+		context.closePath();
+		context.fill();
+		context.lineWidth = 10;
+		context.strokeStyle = '#'+Math.floor(Math.random()*16777215).toString(16);
+		context.stroke();
+		context.scale(1,1);
+
+		window.requestAnimationFrame(draw);
```

Comment:
Be careful with this method, it's not fully supported by all modern browsers. You might need a workaround in a real project.